txnsync/encodedgroupsunmarshalers.go

```diff
@@ -0,0 +1,644 @@
+		return nextSlice(&stub.Receiver, signedTxns[i].Txn.Receiver[:], crypto.DigestSize)
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskAmount.iterate(int(stub.TotalTransactionsCount), len(stub.Amount), func(i int, index int) error {
+		signedTxns[i].Txn.Amount = stub.Amount[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskCloseRemainderTo.iterate(int(stub.TotalTransactionsCount), len(stub.CloseRemainderTo)/crypto.DigestSize, func(i int, index int) error {
+		return nextSlice(&stub.CloseRemainderTo, signedTxns[i].Txn.CloseRemainderTo[:], crypto.DigestSize)
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (stub *txGroupsEncodingStub) reconstructAssetConfigTxnFields(signedTxns []transactions.SignedTxn) (err error) {
+	err = stub.BitmaskConfigAsset.iterate(int(stub.TotalTransactionsCount), len(stub.ConfigAsset), func(i int, index int) error {
+		signedTxns[i].Txn.ConfigAsset = stub.ConfigAsset[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return stub.reconstructAssetParams(signedTxns)
+}
+
+func (stub *txGroupsEncodingStub) reconstructAssetParams(signedTxns []transactions.SignedTxn) (err error) {
+	err = stub.BitmaskTotal.iterate(int(stub.TotalTransactionsCount), len(stub.Total), func(i int, index int) error {
+		signedTxns[i].Txn.AssetParams.Total = stub.Total[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskDecimals.iterate(int(stub.TotalTransactionsCount), len(stub.Decimals), func(i int, index int) error {
+		signedTxns[i].Txn.AssetParams.Decimals = stub.Decimals[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskDefaultFrozen.iterate(int(stub.TotalTransactionsCount), int(stub.TotalTransactionsCount), func(i int, index int) error {
+		signedTxns[i].Txn.AssetParams.DefaultFrozen = true
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskUnitName.iterate(int(stub.TotalTransactionsCount), len(stub.UnitName), func(i int, index int) error {
+		signedTxns[i].Txn.AssetParams.UnitName = stub.UnitName[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskAssetName.iterate(int(stub.TotalTransactionsCount), len(stub.AssetName), func(i int, index int) error {
+		signedTxns[i].Txn.AssetParams.AssetName = stub.AssetName[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskURL.iterate(int(stub.TotalTransactionsCount), len(stub.URL), func(i int, index int) error {
+		signedTxns[i].Txn.AssetParams.URL = stub.URL[index]
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	err = stub.BitmaskMetadataHash.iterate(int(stub.TotalTransactionsCount), len(stub.MetadataHash)/crypto.DigestSize, func(i int, index int) error {
+		return nextSlice(&stub.MetadataHash, signedTxns[i].Txn.AssetParams.MetadataHash[:], crypto.DigestSize)
```

Comment:
I think that I'd rather adding a constant for that, plus making a unit test that verifies the constant is equal to the above length.

Reply:
Why? I'm sure the compiler will make `len(basics.AssetParams.MetadataHash)` a number. The code will also stay correct in case `basics.AssetParams.MetadataHash` changes size.